qemu/tests/qemu_guest_agent.py

```diff
@@ -1026,6 +1027,124 @@ def gagent_check_with_fsfreeze(self, test, params, env):
+            :param input: data to be passed to process stdin (base64 encoded)
+            :param capture_output: bool flag to enable capture of stdout/stderr
+                                   of running process,defaults to false.
+            :return: result of guest-exec cmd
+            """
+
+            # change cmd_args to be a list needed by guest-exec.
+            if cmd_args:
+                cmd_args = cmd_args.split()
+            ret = self.gagent.guest_exec(path=guest_cmd, arg=cmd_args,
+                                         env=env, input_data=input,
+                                         capture_output=capture_output)
+            end_time = time.time() + float(params["guest_cmd_timeout"])
+            while time.time() < end_time:
+                result = self.gagent.guest_exec_status(ret["pid"])
+                if "exited" not in result:
+                    test.fail("The result of guest-exec is not correct.")
+                if result["exited"]:
+                    logging.info("Guest cmd is finished.")
+                    break
+                time.sleep(5)
+
+            if not result["exited"]:
+                test.error("Guest cmd is still running, pls login guest to"
+                           " handle it or extend your timeout.")
+            # check the exitcode and output/error data result
+            if result["exitcode"] == 0:
+                if "out-data" in result:
+                    out_data = base64.b64decode(result["out-data"]).decode()
+                    logging.info("The guest cmd is executed successfully,"
+                                 "the output is: \n %s." % out_data)
+                elif "err-data" in result:
+                    test.fail("When exitcode is 0, should not get error data.")
+            else:
+                if "out-data" in result:
+                    test.fail("When exitcode is 1, should not get output data.")
+                elif "err-data" in result:
+                    err_data = base64.b64decode(result["err-data"]).decode()
+                    logging.info("The guest cmd failed,"
+                                 "the error info is: \n %s" % err_data)
+            return result
+
+        session = self._get_session(params, self.vm)
+        self._open_session_list.append(session)
+
+        error_context.context("Change guest-exec related cmd to white list.",
+                              logging.info)
+        self._change_bl(session)
+
+        guest_cmd = params["guest_cmd"]
+        guest_cmd_args = params["guest_cmd_args"]
+
+        error_context.context("Execute guest cmd and get the output.",
+                              logging.info)
+        result = _guest_cmd_run(guest_cmd=guest_cmd, cmd_args=guest_cmd_args,
+                                capture_output=True)
+
+        if "out-data" not in result and "err-data" not in result:
+            test.fail("There is no output in result.")
```

Comment:
Is that what we expect? If I understand the case correctly, we don't want an `err-data`, what we expect is `result["exitcode"] == 0` and `"out-data" in result`

Reply:
If capture_output=True, what we want is err-data or out-data which depends on the guest cmd. The check point doesn't have "exitcode"